src/PartialCleanDirs.php

```diff
@@ -0,0 +1,248 @@
+     * Filesystem component.
+     *
+     * @var \Symfony\Component\Filesystem\Filesystem
+     */
+    protected $fs;
+
+    /**
+     * What the sort the items in the directories by.
+     *
+     * @var string|\Closure
+     */
+    protected $sort;
+
+    /**
+     * Creates a new PartialCleanDir task.
+     *
+     * @param array $dirs
+     *   The directories to clean. Either an array of directories or an array
+     *   keyed by directory with the number of items to keep within this
+     *   directory as value.
+     * @param null|\Symfony\Component\Finder\Finder $finder
+     *   The finder used to get the files from the source folder.
+     * @param null|\Symfony\Component\Filesystem\Filesystem $fs
+     *   Filesystem component to manipulate files.
+     */
+    public function __construct(array $dirs, $finder = null, $fs = null)
+    {
+        $this->dirs($dirs);
+        $this->finder = is_null($finder)
+            ? (new Finder())->files()->ignoreDotFiles(false)->ignoreVCS(false)
+            : $finder;
+        $this->fs = is_null($fs)
+            ? new Filesystem()
+            : $fs;
+        $this->sort = static::SORT_NAME;
+    }
+
+    /**
+     * Set the finder to use.
+     *
+     * @param Finder $finder
+     *
+     * @return $this
+     */
+    public function finder(Finder $finder)
+    {
+        $this->finder = $finder;
+
+        return $this;
+    }
+
+    /**
+     * Set the filesystem component to use.
+     *
+     * @param Filesystem $fs
+     *
+     * @return $this
+     */
+    public function fileSystem(Filesystem $fs)
+    {
+        $this->fs = $fs;
+
+        return $this;
+    }
+
+    /**
+     * Add directories to clean.
+     *
+     * @param array $dirs
+     *   Either an array of directories or an array keyed by directory with the
+     *   number of items to keep within this directory as value.
+     *
+     * @return $this
+     */
+    public function dirs(array $dirs)
+    {
+        foreach ($dirs as $k => $v) {
+            if (is_numeric($v)) {
+                $this->dir($k, $v);
+                continue;
+            }
+            $this->dir($v);
+        }
+
+        return $this;
+    }
+
+    /**
+     * Add a directory to clean.
+     *
+     * @param $dir
+     *   The directory to clean.
+     * @param $keep
+     *   The number of items to keep in this directory. Defaults to 5.
+     *
+     * @return $this
+     */
+    public function dir($dir, $keep = 5)
+    {
+        $this->dirs[$dir] = $keep;
+
+        return $this;
+    }
+
+    /**
+     * What to sort the folder items by. The last x items (as set by the dir()
+     * method) returned after the sort method will be kept.
+     *
+     * @param string|\Closure $sort
+     *   One of the PartialCleanDir::SORT_* constants or an anonymous function. The
+     *   anonymous function receives two \SplFileInfo instances to compare.
+     *
+     * @return $this
+     */
+    public function sortBy($sort = self::SORT_NAME)
```

Comment:
Symfony Finder doesn't support an order direction, but you can give your own \Closure to this method to define your own sorting algorithm. We could use [iterator_to_array()](http://php.net/manual/en/function.iterator-to-array.php) and [array_reverse()](http://php.net/manual/en/function.array-reverse.php) I guess, but I'm not sure of the performance impact (array_reverse creates a copy of the array and Symfony finder is already pretty clear on the performance impact of sorting files: "This can be slow as all the matching files and directories must be retrieved for comparison.". Reversing the array will not have that same kind of impact of course... 

This is also the reason why I chose to keep the **last** x items after sorting. Usually you'd want to keep the most recent items. Sorting by default is ascending.

An other option is to provide our own sorting functions which sort in reverse order to start with, without using array_reverse, which would have the same performance impact as the already existing sorting functions provided by Symfony Finder